### PR TITLE
oembed updated , spotify url issue is now fixed by removing the trailing slash in the oembed spotify definition

### LINKED ISSUE #10942: Spotify oembed request gets 308 redirect response, and does not follow redirect
### Issue Summary

When you try and embed a spotify URL, urllib gets a 308 response from spotify and raises an exception.

### Steps to Reproduce

1. try and embed a spotify URL, for example https://open.spotify.com/playlist/37i9dQZF1DWUZ5bk6qqDSy
2. a validation error occurs ("Cannot find an embed for this URL.")

By looking at the response body, I have found that it could be fixed by removing the trailing slash in the oembed spotify definition. 

So adding this workaround to my settings fixes it for me:

```python
_spotify = {
    # Only change: removed trailing slash (was "https://embed.spotify.com/oembed/"):
    "endpoint": "https://embed.spotify.com/oembed",
    "urls": [
        r"^https?://open\.spotify\.com/.+$",
        r"^https?://spoti\.fi/.+$",
    ],
}

WAGTAILEMBEDS_FINDERS = [
    {
        'class': 'wagtail.embeds.finders.oembed',
        'providers': [_spotify],
    },
    {
        'class': 'wagtail.embeds.finders.oembed',
    }
]
```

I can make a PR for this if that would be useful.

The other way you could fix this is by making it follow redirects when trying to do an embed, I suppose.

### Technical details

-   Python version: 3.10.8
-   Django version: 4.2.3
-   Wagtail version: 5.0.2

### PR DESCRIPTION
**Pull request by dhruvpal05**
Fixes #10942

_Please check the following:_

-   [ ] Do the tests still pass?[^1]
-   [ ] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/admin/views/account.py [hunk 1]
```diff
@@ -138,10 +138,18 @@ def render(self):
 
 class NameEmailSettingsPanel(BaseSettingsPanel):
     name = "name_email"
-    title = gettext_lazy("Name and Email")
+    # title = gettext_lazy("Name and Email")
```
Comment (RealOrangeOne) [inline wagtail/admin/views/account.py:141]: Issue: Looks like you have some unrelated changes in this PR

**Review by RealOrangeOne — COMMENTED** [review]
Nice catch - this seems to work as expected. It looks like Spotify's `oembed` URL specifically _doesn't_ have a trailing slash:

```
$ curl -I https://embed.spotify.com/oembed/
HTTP/2 308
server: envoy
date: Wed, 27 Sep 2023 16:57:41 GMT
location: /oembed
...
```

File: wagtail/admin/views/account.py (hunk 1 above)
Comment (dhruvpal05) [reply wagtail/admin/views/account.py:141]: These are related to [#10937](https://github.com/wagtail/wagtail/issues/10937) this issue

Comment (laymonage) [reply wagtail/admin/views/account.py:141]: Then it should be included in a separate PR, not combined in this one.

**Comment by lb-** [comment]
Please also review this comment. https://github.com/wagtail/wagtail/issues/10942#issuecomment-1738099993

It may be better to align with the latest documented endpoint.

**Comment by dhruvpal05** [comment]
@laymonage @RealOrangeOne review it please.

File: wagtail/admin/views/account.py (hunk 1 above)
Comment (lb-) [reply wagtail/admin/views/account.py:141]: @dhruvpal05 this PR should not have any of these changes to the account file, please can you either revert this commit or reset this file to it's original state and push.

File: wagtail/embeds/oembed_providers.py [hunk 2]
```diff
@@ -520,8 +520,9 @@
 
 
 spotify = {
-    "endpoint": "https://embed.spotify.com/oembed/",
+    "endpoint": "https://open.spotify.com/oembed",
     "urls": [
+        r"^https?://open\.spotify\.com/episode/7makk4oTQel546B0PZlDM5",
```
Comment (lb-) [inline wagtail/embeds/oembed_providers.py:525]: This specific URL should not be added here. 
```suggestion
```

**Comment by lb-** [comment]
Thanks @dhruvpal05 - there's still some clean up needed on this.

File: wagtail/embeds/oembed_providers.py (hunk 2 above)
Comment (dhruvpal05) [reply wagtail/embeds/oembed_providers.py:525]: ok

**Review by lb- — APPROVED** [review]
@dhruvpal05  as these changes are now conflicting I have just rebased locally and removed those unwanted commits.

Just a reminder for future to keep PRs focused on one thing, it's helpful to create a unique branch for each set of changes you want to do.

We call out this and other helpful tips on our contributor's guide here https://docs.wagtail.org/en/stable/contributing/first_contribution_guide.html#create-a-fresh-branch-for-your-contributions

Nonetheless, thank you so much for finding a fix here and responding to other feedback about the URL changes.

I have validated this locally and it's working great.  I will force push to your branch and merge in shortly.

Screenshot 2023-10-02 at 7 54 43 am

**Comment by lb-** [comment]
The only name I could find was `Dhrűv`, please let us know if you want a different name listed. Thanks for making your first contribution to Wagtail!

**Comment by dhruvpal05** [comment]
@lb-  Thankyou for the changes and guidance, will surely take care of creating a unique branch for each set of changes. Dhruv is fine.
